rir/src/compiler/pir/closure.cpp

```diff
@@ -54,7 +54,7 @@ ClosureVersion* Closure::cloneWithAssumptions(ClosureVersion* version,
 ClosureVersion*
 Closure::findCompatibleVersion(const OptimizationContext& ctx) const {
     // ordered by number of assumptions
-    for (auto c = versions.begin(); c != versions.end(); c++) {
+    for (auto c = versions.rbegin(); c != versions.rend(); c++) {
         auto candidate = *c;
```

Comment:
I'm noticing now that these are probably making unnecessary copies. The copies are small, but still...

So you'll want `const auto& candidate = *c;` and `const auto& candidateCtx = candidate.first;`.